resolver/resolver_test.go

```diff
@@ -481,3 +297,179 @@ func TestLaunchZK(t *testing.T) {
 		// expected
 	}
 }
+
+func message(asserts ...func(*dns.Msg) []error) func(*dns.Msg) []error {
```

Comment:
why call it message? messageAssertions would be more verbose and actually explain that it's about assertions.